src/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector/EventLogDataCollector.cs

```diff
@@ -149,12 +147,13 @@ internal EventLogDataCollector(IFileHelper fileHelper)
     /// Used by the data collector to send warnings, errors, or other messages
     /// </param>
     /// <param name="dataCollectionEnvironmentContext">Provides contextual information about the agent environment</param>
+    [MemberNotNull(nameof(_events), nameof(_dataSink), nameof(_logger))]
     public override void Initialize(
-        XmlElement configurationElement,
+        XmlElement? configurationElement,
         DataCollectionEvents events!!,
         DataCollectionSink dataSink!!,
         DataCollectionLogger logger!!,
-        DataCollectionEnvironmentContext dataCollectionEnvironmentContext)
+        DataCollectionEnvironmentContext dataCollectionEnvironmentContext!!)
```

Comment:
This is a change of behavior but we weren't doing null check so if `Initialize` is called with null for this arg we would fail with NRE.